homeassistant/components/light/osramlightify.py

```diff
@@ -50,20 +58,26 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
                                                                      str(err))
             _LOGGER.exception(msg)
             return False
-        setup_bridge(bridge, add_devices)
+        setup_bridge(bridge, add_devices, add_groups)
     else:
         _LOGGER.error('No host found in configuration')
         return False
 
 
-def setup_bridge(bridge, add_devices_callback):
+def setup_bridge(bridge, add_devices_callback, add_groups):
     """Setup the Lightify bridge."""
     lights = {}
 
     @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
     def update_lights():
         """Update the lights objects with latest info from bridge."""
-        bridge.update_all_light_status()
+        try:
+            bridge.update_all_light_status()
+            bridge.update_group_list()
+        except socket.error:
+            errno, errstr = sys.exc_info()[:2]
```

Comment:
Wait, what are you doing here? You can just catch the actual error instead of hacking around it using exception frames:

```python
except socket.error as err:
    if isinstance(err, socket.timeout):
        _LOGGER.error(…)
```

However, you are now swallowing all the not timeout related errors which means that something might go wrong but no one will know.